egs/wsj/s5/steps/dur_model/train.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+
+# Copyright 2015 Hossein Hadian
+
+num_epochs=20
+minibatch_size=512
+num_epochs_per_iter=1
+nnet_config=
+early_stop=true
+early_stop_min_epochs=5
+cmd=run.pl
+use_gpu=true    # if true, we run on GPU.
+stage=-2
+shuffle_buffer_size=5000
+egs_opts=
+nnet_opts=
+max_duration=0
+left_context=4
+right_context=2
+min_repeat_count=5
+
+echo "$0 $@"  # Print the command line for logging
+
+[ -f ./path.sh ] && . ./path.sh; # source the path.
+. parse_options.sh || exit 1;
+
+if [[ $# != 3 ]]; then
+   echo "Usage: $0 [options] <phones-dir> <ali-dir> <duration-model-dir>"
+   echo "e.g.: $0 data/lang/phones exp/mono_ali exp/mono/durmod"
+   echo ""
+   echo "Main options (for others, see top of script file):"
+   echo "  --num-epochs <number>                       # max number of epochs for training"
+   echo "  --minibatch-size <size>                     # minibatch size"
+   echo "  --compute-prob-interval <int>               # interval for measuring accuracy (diagnostics)"
+   echo "  --nnet-config <nnet3-conf-file>             # use this config for training"
+   echo "Options related to initializing (stage 0):"
+   echo "  --max-duration <duration-in-frames>         # max duration; if not set, it will be determined automatically"
+   echo "  --left-context <size>                       # left phone context size"
+   echo "  --right-context <size>                      # right phone context size"
+   exit 1;
+fi
+
+phones_dir=$1
+alidir=$2
+dir=$3
+
+durmodel=$dir/durmodel.mdl
+mkdir -p $dir/log
+
+if [ $stage -le -2 ]; then
+  echo "$0: Initializing the duration model..."
+
+  transmodel=$alidir/final.mdl
+  for f in $transmodel $phones_dir/roots.int $phones_dir/extra_questions.int; do
+    [ ! -f $f ] && echo "$0: Required file for initializing not found: $f" && exit 1;
+  done
+
+  if [ $max_duration == 0 ]; then
+    echo "$0: Determining max-duration..."
+    if [ ! -f $dir/all.ali ]; then
+      gunzip -c $alidir/ali.*.gz >> $dir/all.ali || exit 1;
+    fi
+
+    max_duration=`ali-to-phones --write-lengths $transmodel ark:$dir/all.ali \
```

Comment:
you can directly use 'gunzip -c $alidir/ali.*.gz' in the argument rather than creating a temporary file.